### ISSUE TITLE
Feature Request: Add `ignore_using` parameter to `ModuleDoc` check

### ISSUE THREAD

**Issue by pierrelegall** [rrrene/credo#1255]
# Feature Request: Add `ignore_using` parameter to `ModuleDoc` check

## Summary

Add an `:ignore_using` parameter to `Credo.Check.Readability.ModuleDoc` that suppresses missing `@moduledoc` warnings for modules that `use` a given module.

## Motivation

The existing `:ignore_names` parameter relies on module naming conventions to decide what to skip. This is brittle for projects where module names don't follow predictable patterns. For example, a Phoenix LiveView module named `MyApp.Admin.Dashboard` won't match any name-based regex, yet it clearly doesn't need a `@moduledoc`.

Being able to write `ignore_using: [Phoenix.LiveView]` and have it apply to all modules that `use Phoenix.LiveView` — regardless of their name — would be more robust and easier to maintain.

## Proposed Behavior

```elixir
# .credo.exs
{Credo.Check.Readability.ModuleDoc, [
  ignore_using: [Phoenix.LiveView, "MyApp.Web"]
]}
```

The parameter should accept atoms, strings, and regexes, consistent with `:ignore_names`.

**Comment by pierrelegall** [comment]
Solved by #1254.

But it's named `ignore_modules_using` instead of `ignore_using` like proposed in this issue.

We can notice the previously existing `ignore_names` is not verbose (we implicitly talk about modules) and `ignore_modules_using` is verbose (more explicit here). Just to notice this choice could be surprising for the users to have two different naming strategies here.
